src/cloudant/result.py

```diff
@@ -228,16 +227,36 @@ def __getitem__(self, key):
 
     def __iter__(self):
         """
-        Iteration Support for large views
-
-        Uses skip/limit to consume a view in chunks controlled
-        by the page_size setting and retrieves a batch of records
-        from the view or result and then yields each element.
-
-        Since this uses skip/limit to perform the iteration, they
-        cannot be used as optional arguments to the result, but startkey
-        and endkey etc can be used to constrain the result of the iterator
-
+        Provides iteration support, primarily for large data collections.
+        The iterator uses skip/limit to consume data in chunks controlled
+        by the ``page_size`` setting and retrieves a batch of data
+        from the result collection and then yields each element.  Since the
+        iterator uses skip/limit to perform the iteration, ``skip`` and
+        ``limit`` cannot be used as optional arguments to customize the result,
+        but other optional arguments can be used to customize the result
+        of the iterator.
+
+        See :class:`~cloudant.result.Result` for Result iteration examples.
+
+        Optional arguments:
+
+        :param bool descending: Valid optional argument
+        :param endkey: alid optional argument.  Can be
```

Comment:
Removed the whole list of "optional arguments" and replaced them with a reference to the View.make_result() method docs.  Arguments are attached to the Result and not passed in to the __iter__ method so it was misleading to have them here in the first place.  Done in a2556617e0362fd5161b8bde868790b0de555b97.